tests/router_test.js

```diff
@@ -1594,6 +1594,27 @@ scenarios.forEach(function(scenario) {
       showPost: showPostHandler,
     };
 
+    // Check for mid-transition correctness.
```

Comment:
Can you move this into a test of its own? Seems odd to just be an extra assertion in a somewhat unrelated test...